internal/telemetry/logs/service_logs_test.go

```
@@ -0,0 +1,55 @@
+	if strings.Contains(s, `attributes['deployment_environment']`) {
+		t.Errorf("env filter must not use attributes['deployment_environment'] (wrong namespace + underscore key), got: %s", s)
+	}
+}
+
+// TestAddServiceLogsEnvFilter_ValueIsPreserved ensures the env value is present in
+// the generated filter condition.
+func TestAddServiceLogsEnvFilter_ValueIsPreserved(t *testing.T) {
+	base := buildServiceLogsQuery("svc", nil, nil)
+	result := addServiceLogsEnvFilter(base, "staging")
+
+	raw, _ := json.Marshal(result)
```

golangci-lint (2.12.2)

[error] Error return value of `encoding/json.Marshal` is not checked: unsafe type `interface{}` found

(errchkjson)